### PR TITLE
Compute numpy dtype

### PR DESCRIPTION
**Pull request by vallsv**
This PR introduce a `dtype` method for definition classes.

This can be used with numpy to parse a bytes structure.

This can be used the following way:
```
import pycstruct
import numpy

rgba_t = pycstruct.StructDef()
rgba_t.add("uint8", "r")
rgba_t.add("uint8", "g")
rgba_t.add("uint8", "b")
rgba_t.add("uint8", "a")

data = b'\x01\x02\x03\x00'
rgba = numpy.frombuffer(data, dtype=type_t.dtype(), count=1)[0]
assert rgba["r"] == 1
```
This also can be deserialized the following way to read the fields by argument.
```
data = b'\x01\x02\x03\x00'
rgba = numpy.frombuffer(data, dtype=type_t.dtype(), count=1)[0] 
rgba = numpy.rec.array(rgba) 
assert rgba.r == 1
```

There is some limitations:
- No bitfields
- <s>No unions</s> (fixed)
- No string decoding (everything is bytes)
- <s>The result have to be handled in different way sometimes</s> (fixed)

### CODE REVIEW

File: pycstruct/pycstruct.py [hunk 1]
```diff
@@ -460,6 +476,42 @@ def create_empty_data(self):
     buffer = bytearray(self.size())
     return self.deserialize(buffer)
 
+  def dtype(self):
```
Comment (midstar) [inline pycstruct/pycstruct.py]: Could you add some comments for this method. I would like the comment to include:

- What it does (i.e. used for serialization/deserialization using numpy structured arrays)
- Why to use it (I guess performance is the main reason)
- Limitations (Nested structs, enums etc)
- Some examples on how to use it

You can check that the documentation looks ok by entering the doc directory and write "make html".

**Review by midstar — CHANGES_REQUESTED** [review]
Check my comment regarding documentation. Also, it seems to be a merge conflict right now.

**Comment by vallsv** [comment]
Wow you did a lot of work this days.

I did rebase changes + some documentation.

Let me know if it is enough.

I could also add a serialization example, i never tried it, but i am not so sure it is useful.

**Comment by vallsv** [comment]
I feel like pylint is a bit too strong. I have no idea what it complains about.

**Comment by vallsv** [comment]
I did found the problem, but it sounds weird.
The code is better this way anyway.

**Review by midstar — APPROVED** [review]

**Pull request merged**